File: c++/1935.cpp
#include <iostream>
#include <map>
#include <string>
#include <vector>

using namespace std;

map<char, int> dic;
vector<double> stack;

int main() {
  freopen("example.txt", "r", stdin);

  int N;
  cin >> N;

  string s;
  cin >> s;

  for (int i = 0; i < N; i++) {
    int value;
    cin >> value;
    dic[char('A' + i)] = value;
  }

  for (int i = 0; i < s.length(); i++) {
    if (s[i] == '*' || s[i] == '+' || s[i] == '/' || s[i] == '-') {
      double b = stack.back();
      stack.pop_back();
      double a = stack.back();
      stack.pop_back();
      double result;

      if (s[i] == '*') {
        result = a * b;
      } else if (s[i] == '+') {
        result = a + b;
      } else if (s[i] == '/') {
        result = a / b;
      } else if (s[i] == '-') {
        result = a - b;
      }
      stack.push_back(result);
    } else {
      stack.push_back(dic[s[i]]);
    }
  }

  cout << fixed;
  cout.precision(2);

  cout << stack[0];

  return 0;
}

File: js/1935.js
const fs = require('fs');

const inputs = fs.readFileSync('example.txt').toString().trim().split('\n');

const dic = {};
const N = +inputs[0];
const s = inputs[1];
const stack = [];

for (let i = 0; i < N; i++) {
  const a = String.fromCharCode('A'.charCodeAt(0) + i);
  dic[a] = +inputs[i + 2];
}

for (let i = 0; i < s.length; i++) {
  if (s[i] === '*' || s[i] === '+' || s[i] === '/' || s[i] === '-') {
    const a = stack.pop();
    const b = stack.pop();
    let result;

    if (s[i] === '*') result = a * b;
    else if (s[i] === '+') result = a + b;
    else if (s[i] === '/') result = b / a;
    else if (s[i] === '-') result = b - a;

    stack.push(result);
  } else {
    stack.push(dic[s[i]]);
  }
}

console.log(stack[0].toFixed(2));
